internal/dag/builder.go

```diff
@@ -534,11 +534,17 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 					return
 				}
 
-				sec, err = b.lookupSecret(k8s.FullName{Name: b.FallbackCertificate.Name, Namespace: b.FallbackCertificate.Namespace}, validSecret)
+				sec, err = b.lookupSecret(*b.FallbackCertificate, validSecret)
 				if err != nil {
 					sw.SetInvalid("Spec.Virtualhost.TLS fallback certificate Secret %q is invalid: %s", b.FallbackCertificate, err)
 					return
 				}
+
+				if !b.delegationPermitted(*b.FallbackCertificate, proxy.Namespace) {
+					sw.SetInvalid("Spec.VirtualHost.TLS fallback secret %s is not configured for certificate delegation to namespace %s", b.FallbackCertificate.Name, b.FallbackCertificate.Namespace)
```

Comment:
The problem is that the certificate isn't delegated, not that it's not permitted to be delegated. So the suggestion doesn't read properly as to what's wrong.

Reply:
Since the problem here is identical to the delegation check for regular certificates, we should align the status message. If you prefer your more specific message, then I'd suggest updating both occurrences. Just please use %q (which clarifies the distinction between object names and message test), and check that we are consistently capitalizing Kubernetes kind names. I know this is super picky, but I think that it's worth trying to make the error status message as clear and consistent as we can.

Reply:
I dislike %q because you get \"log messages\" like \"this\".

Reply:
Because it quotes in the JSON output? In general, JSON logs work better when tooling consumes them I think :)

In this case, it's a status field. The %q makes it clearer to the person reading this that its refers to a object name and that it's not part of the message text. If the string is ever empty, %q makes it clear that there is an empty string. So I get that the quoting looks awkward in JSON, but because this is the only place where we tell the user about errors, I'd argue that reducing any ambiguity as much as possible makes the quoting worth it.

Reply:
I like the use of %q because of how it shows the empty string case, although I agree the quoting is awkward when it's JSON. This should be mitigated as we move to Conditions anyway.

Reply:
cool cool I updated to use the %q